packages/notebook/src/actions.tsx

```diff
@@ -652,7 +659,10 @@ export namespace NotebookActions {
       notebook.activeCellIndex++;
     }
 
-    void Private.handleRunState(notebook, state, true);
+    // If a cell is outside of viewport and scrolling is needed,
+    // prefer to align the cell to the top viewport edge,
+    // rather than to the bottom (so that editor is visible)
+    void Private.handleRunState(notebook, state, true, 'start');
```

Comment:
This is leading to the `should scroll as little as possible on markdown rendering` test failing, which highlights trade-offs of different approaches, compare:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/cb333ed0-efd1-4cbc-9d53-bfc3cb0bc873)

with alternative:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/5dda701e-2e6a-4680-90ff-0526f1b76f50)

and with:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/e2390e14-1734-4142-996d-e3f82e400b25)

Reply:
It seems to me that we would want to restore the JupyterLab 3 behaviour here, maybe a variation of it (say "top edge in the 1/4 of the viewport from the top").

Reply:
So the problem with the earlier implementation (before https://github.com/jupyterlab/jupyterlab/pull/15288) was not that `auto` alignment differed from Lab 3.6, just that "center" was implemented as "align the middles of the cell and the viewport". This obviously led to very annoying behaviour for cells longer than viewport as seemingly random part of the output was shown:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/a85edd21-1902-4821-89b5-3acceb6d1902)

I will push a commit to fix that behaviour, which should simplify the code too.